functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -76,101 +81,74 @@ public String createWorkspace(String pathToJson, String token) throws Exception
-            .append(path)
-            .append("?")
-            .append("masterUrl=")
-            .append(this.host)
-            .append("&")
-            .append("namespace=sth");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Content-Type", "application/json");
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    Request request = requestBuilder.post(body).build();
-    OkHttpClient client = new OkHttpClient();
+    JsonObject body = PARSER.parse(json).getAsJsonObject();
+    String relativePath = "/workspace";
+    String authorization = "Bearer " + token;
+    HashMap<String, String> queryParameters = new HashMap<>();
+    queryParameters.put("masterUrl", this.host);
+    queryParameters.put("namespace", "sth");
     try {
-      Response response = client.newCall(request).execute();
-      return getNameFromResponse(response);
-    } catch (IOException e) {
-      LOG.error("Workspace could not be created : " + e.getMessage(), e);
-      return null;
+      return getNameFromResponse(
+          cheStarterClient.sendRequest(
+              relativePath, HttpMethods.POST, body, authorization, queryParameters.entrySet()));
+    } catch (Exception e) {
+      LOG.error("Get name from response failed with exception:" + e.getMessage(), e);
+      throw new RuntimeException("Failed to parse createWorkspace response.", e);
     }
   }
 
-  public boolean deleteWorkspace(String workspaceName, String token) {
-    StringBuilder sb = new StringBuilder(this.cheStarterURL);
-    String path = "/workspace/" + workspaceName;
-    sb.append(path);
-    sb.append("?");
-    sb.append("masterUrl=").append(this.host).append("&").append("namespace=sth");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Content-Type", "application/json");
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    OkHttpClient client = new OkHttpClient();
-    try {
-      Response response = client.newCall(requestBuilder.delete().build()).execute();
-      LOG.info("Workspace delete response : " + response.message());
-      if (!response.isSuccessful()) {
-        if (response.message().equals("Workspace not found")) {
-          LOG.warn(
-              "Workspace could not be deleteded because workspace is not found. Continuing tests.");
-          return true;
-        }
+  public boolean deleteWorkspace(String workspaceName, String token) throws IOException {
+    String relativePath = "/workspace/" + workspaceName;
+    HashMap<String, String> queryParameters = new HashMap<>();
+    queryParameters.put("masterUrl", this.host);
+    queryParameters.put("namespace", "sth");
+    String authorization = "Bearer " + token;
+    Response response =
+        cheStarterClient.sendRequest(
+            relativePath, HttpMethods.DELETE, null, authorization, queryParameters.entrySet());
+    if (response.isSuccessful()) {
```

Comment:
Shouldn't here be if(!response.isSuccessfull()) ?

Reply:
fixed, added into test case.